### PR TITLE
Fix importing for VGG Face 2

### LINKED ISSUE #337: Import VGG Face2 dataset
### Steps to reproduce problem
1. Download the original VGG Face2 dataset;
2. Import the loaded dataset into a Datumaro project.

### Current behaviour
Error with message:
```bash
Traceback (most recent call last):
  File "/home/ksizov/d2/datumaro/venv/bin/datum", line 33, in <module>
    sys.exit(load_entry_point('datumaro', 'console_scripts', 'datum')())
  File "/home/ksizov/d2/datumaro/datumaro/cli/__main__.py", line 126, in main
    return args.command(args)
  File "/home/ksizov/d2/datumaro/datumaro/cli/contexts/source.py", line 169, in add_command
    project.make_source_project(name).make_dataset()
  File "/home/ksizov/d2/datumaro/datumaro/components/project.py", line 425, in make_dataset
    return ProjectDataset(self)
  File "/home/ksizov/d2/datumaro/datumaro/components/project.py", line 63, in __init__
    sources[s_name] = Dataset.import_from(url,
  File "/home/ksizov/d2/datumaro/datumaro/components/dataset.py", line 815, in import_from
    extractors.append(env.make_extractor(
  File "/home/ksizov/d2/datumaro/datumaro/components/environment.py", line 274, in make_extractor
    return self.extractors.get(name)(*args, **kwargs)
  File "/home/ksizov/d2/datumaro/datumaro/plugins/vgg_face2_format.py", line 37, in __init__
    self._items = list(self._load_items(path).values())
  File "/home/ksizov/d2/datumaro/datumaro/plugins/vgg_face2_format.py", line 94, in _load_items
    raise Exception("Item %s: an image can have only one "
Exception: Item 0001_01: an image can have only one set of landmarks
```

### Expected behaviour
Successful importing.

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Summary
Fixed #337.

### How to test
Add the original VGG Face 2 (train subset) into a Datumaro project.
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by kirill-sizov** [comment]
So I fixed the bug with `item.id` for VGG Face 2. But now there is another problem: the original dataset contain different categories for test and train subsets (error: `Merging of datasets with different categories is only allowed in 'merge' command.`)

@zhiltsov-max, should we support such cases?

**Comment by zhiltsov-max** [comment]
If a dataset is expected to contain different categories, which can't be simply merged, then importer should import only one task / subset (and support importing by a direct path). Check coco importer.

File: datumaro/plugins/vgg_face2_format.py [hunk 1]
```diff
@@ -23,20 +23,29 @@ class VggFace2Path:
     IMAGES_DIR_NO_LABEL = 'no_label'
 
 class VggFace2Extractor(SourceExtractor):
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:25]: If extractor outputs multiple subsets, it must be inherited from `Extractor`.

File: datumaro/plugins/vgg_face2_format.py [hunk 2]
```diff
@@ -22,21 +22,30 @@ class VggFace2Path:
...
-            if subset.startswith(VggFace2Path.LANDMARKS_FILE):
-                subset = subset.split('_')[2]
-        super().__init__(subset=subset)
+        self._subsets = {osp.splitext(f.split('_')[2])[0] for f in annotation_files}
 
         self._categories = self._load_categories()
-        self._items = list(self._load_items(path).values())
+        self.items = []
+        for subset in self._subsets:
+            self._items.extend(list(self._load_items(subset).values()))
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:48]: I suppose that one of this variables is misspelled.

File: datumaro/plugins/vgg_face2_format.py [hunk 3]
```diff
@@ -130,15 +141,21 @@ def _split_item_path(path):
...
     @classmethod
     def find_sources(cls, path):
-        return cls._find_sources_recursive(path, '.csv', 'vgg_face2',
-            dirname=VggFace2Path.ANNOTATION_DIR,
-            file_filter=lambda p: \
-                not osp.basename(p).startswith(VggFace2Path.BBOXES_FILE))
+        annotation_dir = osp.join(path, VggFace2Path.ANNOTATION_DIR)
+        if osp.isdir(annotation_dir):
+            return [{'url': annotation_dir, 'format': 'vgg_face2'}]
+        return []
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:147]: Now it is impossible to load a specific subset.

File: datumaro/plugins/vgg_face2_format.py (hunk 1 above)
Comment (kirill-sizov) [reply datumaro/plugins/vgg_face2_format.py]: Fixed.

File: datumaro/plugins/vgg_face2_format.py (hunk 2 above)
Comment (kirill-sizov) [reply datumaro/plugins/vgg_face2_format.py]: Fixed.

File: datumaro/plugins/vgg_face2_format.py (hunk 3 above)
Comment (kirill-sizov) [reply datumaro/plugins/vgg_face2_format.py]: I pushed new commits, now user can load the vgg_face2 dataset in two ways: 1) from directory with dataset (all subsets will be load) 2) from specific annotation file (only specified subset will be load)

File: datumaro/plugins/vgg_face2_format.py [hunk 4]
```diff
@@ -130,15 +158,27 @@ def _split_item_path(path):
...
-            file_filter=lambda p: \
-                not osp.basename(p).startswith(VggFace2Path.BBOXES_FILE))
+        if osp.isdir(path):
+            annotation_dir = osp.join(path, VggFace2Path.ANNOTATION_DIR)
+            if osp.isdir(annotation_dir):
+                return [{'url': annotation_dir, 'format': 'vgg_face2'}]
+        elif osp.isfile(path):
+            if (osp.basename(path).startswith(VggFace2Path.LANDMARKS_FILE) or \
+                osp.basename(path).startswith(VggFace2Path.BBOXES_FILE)) and \
+                path.endswith('.csv'):
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:168]:
```suggestion
            if (osp.basename(path).startswith(VggFace2Path.LANDMARKS_FILE) or \
                        osp.basename(path).startswith(VggFace2Path.BBOXES_FILE)) and \
                    path.endswith('.csv'):
```

File: datumaro/plugins/vgg_face2_format.py [hunk 5]
```diff
@@ -22,21 +22,44 @@ class VggFace2Path:
...
+                osp.basename(p).startswith(VggFace2Path.LANDMARKS_FILE)) and \
+                p.endswith('csv')]
+
+        if len(annotation_files) < 1:
+            raise Exception("Can't find annotations in the directory '%s'" % path)
+
+        super().__init__()
+
+        self._dataset_dir = osp.dirname(self._path)
+        self._subsets = set([subset]) if subset else set(
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:47]:
```suggestion
        self._subsets = {subset} if subset else set(
```

File: datumaro/plugins/vgg_face2_format.py [hunk 6]
```diff
@@ -22,21 +22,44 @@ class VggFace2Path:
...
+        elif osp.isfile(path):
+            subset = osp.splitext(osp.basename(path).split('_')[2])[0]
+            self._path = osp.dirname(path)
+        else:
+            raise Exception("Can't read annotations from '%s'" % path)
+
+        annotation_files = [p for p in os.listdir(self._path)
+            if (osp.basename(p).startswith(VggFace2Path.BBOXES_FILE) or \
+                osp.basename(p).startswith(VggFace2Path.LANDMARKS_FILE)) and \
+                p.endswith('csv')]
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:39]:
```suggestion
                p.endswith('.csv')]
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**